### DISCUSSION TITLE
フロントエンドの作業の進め方について（相談）

### DISCUSSION THREAD

**Discussion by tenk-9** [ARCircle/Kojirer#34] [General]
各端末（受付端末，調理場端末，呼び出し端末）のフロントエンドの実装に関してですが，作業の進め方についてどのようにしようか迷っており，メンバーの意見を聞きたいです．
考えている方法は次の3案で，メンバーの皆さんはどれがいいと思いますか？また，このほかに案があれば共有してくれると嬉しいです．

### 案 1
- 山口が全ての端末のUIのたたき（画像）を作る
- 各メンバーに実装を割り振り，割り振られた端末のUIを各人で実装する
- 改善点や提案がある場合は，メンバー内で話し合い，調整する

- Pros
  - 山口は飲食店の勤務経験があるため，運用上どのようなUIだと作業しやすいか，大まかなイメージがある
    - UI案（たたき）がまとまるまで時間が短い
    - 飲食の勤務経験が無いと，どのようなUIにすればいいか考えるのが大変と思われ，それを回避できる
  - 端末ごとに実装を行う人がわかれるため，実装の過程におけるコミュニケーションコストが少ない
- Cons
  - 山口がUI案を出すまで，他のメンバーが作業できない（山口がボトルネック）
  - UI案の作成者と実装者が別になるとき，実装する際に少し戸惑うかも
    - できるだけ具体的な指示をすればいいのですが，混乱は避けられないと思われる

### 案 2
- 各メンバーが担当する端末を決め，UI案の作成から実装までを一人で行う
- UI案は実装に移る前に一度メンバー内で共有し，Reviewする
- 改善点や提案がある場合は，メンバー内で話し合い，調整する

- Pros
  - UI案の作成から実装まで，スムーズに作業が進められる（他人がボトルネックになりにくい）
  - UI案作成者と実装者が同一なので，案1で懸念されるような実装上の混乱は生じにくい
  - 案1で存在するUI案作成→実装へ移行する際のコミュニケーションは発生しない
- Cons
  - 飲食店での勤務経験等が無いと，UI案を考えるのは難しい
    - UIが原因となって運用者の作業負担が増えてはいけない
    - どのようなUIが操作しやすいかは，経験がないと判断しづらい面がある

### 案 3
- 取り組むべき端末をフォーカスし，メンバー全員でUI案決めから実装までを行う
- 1つの端末が実装できたら，次の端末へフォーカスを移す

- Pros
  - UI案の設計意図等が全体で共有されるため，UI案と実装の食い違いは生じにくい
  - スケジュール管理がしやすそう
- Cons
  - コミュニケーションコストが莫大
  - 実際にこの方法で作業するとなると，対面で作業を進める必要がありそう

**Comment by tenk-9** [comment]
@4416kidr cc. @woodnext

**Comment by 4416kidr** [comment]
私は案2に賛成します。

案3については指摘にもあるようにコミュニケーションや実装に競合が発生する可能性が高く、個人的にはあまり好ましくないと考えます。案1に関しては山口くんの負担が大きすぎると思うので、あまり好ましくないと考えます。したがって私は案2に賛成します。
ただしUIを考える人と実装をする人は絶対に同じでなくても良いと思います。実装時に戸惑いが発生しうる問題はありますが、それはコミュニケーションによって解消すればいいと思います。

案2の懸念点としてはUI案を各メンバーが別々に考えるので各画面UIの統一感があまりないような状態になってしまいやすいことがあります。この点に関しては最後に全体の統一感をレビューする必要があると思います。

最後に飲食の勤務経験がないとUIを考えるのは難しいのではないかと指摘していますが、その点はあまり考慮しなくても良いと考えます。理由としては最も勤務経験が要求される飲食店内部で使われる厨房のUIが決定しているからです。そのほかのUIに関しては顧客との接点になるため、利用者であっても考えることができると思います。少なくとも私は受付画面と呼び出し画面について簡単なイメージでしたらあります。

**Reply by tenk-9** [reply]
> 案2に賛成します
> UIを考える人と実装をする人は絶対に同じでなくても良いと思います

ありがとうございます、案2で行くことにします。案作成と実装の担当については、確かにコミュニケーションでどうにかなりますね！

> 案2の懸念点としてはUI案を各メンバーが別々に考えるので各画面UIの統一感があまりないような状態になってしまいやすいことがあります。この点に関しては最後に全体の統一感をレビューする必要があると思います。

確かにその通りです。顧客に見せる想定ではない*ので、しっかり統一させなくても大丈夫なように思いますが、ある程度のフォーマットは決めておいたほうが良さそうですね。

> そのほかのUIに関しては顧客との接点になるため

これについて、*とも関連するのですが受付端末については、従業員の操作を想定していました（呼び出し端末については考えがまとまっていないです）
というのも、会計処理等は従業員が行うので従業員が操作した方がスムーズなためです

**Reply by 4416kidr** [reply]
> 確かにその通りです。顧客に見せる想定ではない*ので、しっかり統一させなくても大丈夫なように思いますが、ある程度のフォーマットは決めておいたほうが良さそうですね。

この点については認識に違いがあったみたいです。従業員が操作するのであればデザインの統一についてはそこまでしっかりしなくても良さそうですね。

**Reply by tenk-9** [reply]
従業員が操作する前提として，受付端末のUI案は軽くイメージできますか？

**Reply by 4416kidr** [reply]
あまり変わらないと思うので、イメージできそうです。
